self_supervised/vision/simclr.py

```diff
@@ -22,12 +22,16 @@ def create_simclr_model(encoder, hidden_size=256, projection_size=128, nlayers=2
     apply_init(projector)
     return SimCLRModel(encoder, projector)
 
+# Cell
+@delegates(get_multi_aug_pipelines)
+def get_simclr_aug_pipelines(size, **kwargs): return get_multi_aug_pipelines(n=2, size=size, **kwargs)
+
 # Cell
 class SimCLR(Callback):
     order,run_valid = 9,True
-    def __init__(self, size, aug_func=get_batch_augs, temp=0.07, print_augs=False, **aug_kwargs):
-        self.aug1 = aug_func(size, **aug_kwargs)
-        self.aug2 = aug_func(size, **aug_kwargs)
+    def __init__(self, aug_pipelines=[], temp=0.07, print_augs=False, **aug_kwargs):
```

Comment:
Sry, just figured out how to comment the code directly from GitHub.
This default value: `aug_pipelines=[]`
If the user doesn't provide anything and the default values are used, then an error will be raised.

Reply:
I would make it non-optional:
`aug_pipelines`

Also just an FYI before releasing v1.0.0 you might want to consider adding `def __init__(self, aug_pipelines, *, temp=0.7, ....)` then the user is forced to provide the values after * as keyword arguments which may give us more flexibility in the future, when we decide to rename/add positional only arguments.

Reply:
That makes sense I couldn't decide how to best order the arguments based on importance but I guess it's best to position `aug_pipelines` always as first.

Reply:
> Also just an FYI before releasing v1.0.0 you might want to consider adding def __init__(self, aug_pipelines, *, temp=0.7, ....) then the user is forced to provide the values after * as keyword arguments which may give us more flexibility in the future, when we decide to rename/add positional only arguments.

I haven't seen such convention before, interesting :) 

For now, I am leaving all non-optional parameters without any defaults and remaining parameters are defaulted to the best values from the papers.

Reply:
> For now, I am leaving all non-optional parameters without any defaults and remaining parameters are defaulted to the best values from the papers.

:+1:

Reply:
Thanks again for the quick feedback!

Reply:
Thanks for working at such an incredible speed!
Reviewing the newest changes is becoming part of my daily morning routine. :D